STM32Project/Core/Src/traction_control.c

```diff
@@ -87,6 +97,61 @@ void determineSlippage(){
         // printf("Wheel with the worst slippage is: %d\r\n", WHEEL);
         linearTraction(THROTTLE_INPUT);
     }   
+    else if(CURRENT_STEERING == RIGHT){
+        int inside_rpmA = RPM_VALUES[1];    //front left
+        int inside_rpmB = RPM_VALUES[4];    //back right
+
+        int outisde_rpmA = RPM_VALUES[2];   //front right
+        int outside_rpmB = RPM_VALUES[3];   //back left
+
+        float turn = WHEELBASE /calcTurnAngle(JOYSTICK);
+        float outterRadius = turn + WHEELBASE/2;
+        float innerRadius = turn - WHEELBASE/2;
+
+        float threshMax = (outterRadius/innerRadius) * (1 + STEERING_THRESHOLD);
+        float threshMin = (outterRadius/innerRadius) * (1 - STEERING_THRESHOLD);
+        for(int i = 0; i < sizeof(RPM_VALUES)/sizeof(RPM_VALUES[0]); i++){
+            if(RPM_VALUES[i] > threshMax){
+                if(RPM_VALUES[i] > maxVal){
+                    maxVal = RPM_VALUES[i];
+                    WHEEL = i;
+                }
+                // printf("Slippage Detected: Wheel %d\r\n", i);
+                push(WHEELS_WITHOUT_TRACTION,i);
+            }
+            else{ push(WHEELS_WITH_TRACTION,i); }
+        }
+    linearTraction(THROTTLE_INPUT);
+    }
+    else if(CURRENT_STEERING == LEFT){
+        int inside_rpmA = RPM_VALUES[2];    //front right
+        int inside_rpmB = RPM_VALUES[3];    //back left
+
+        int outisde_rpmA = RPM_VALUES[1];   //front left
+        int outside_rpmB = RPM_VALUES[4];   //back right
+        
+        float turn = WHEELBASE /calcTurnAngle(JOYSTICK);
+        float outterRadius = turn + WHEELBASE/2;
+        float innerRadius = turn - WHEELBASE/2;
+
+        float threshMax = (outterRadius/innerRadius) * (1 + STEERING_THRESHOLD);
+        float threshMin = (outterRadius/innerRadius) * (1 - STEERING_THRESHOLD);
+        
+        for(int i = 0; i < sizeof(RPM_VALUES)/sizeof(RPM_VALUES[0]); i++){
+            if(RPM_VALUES[i] > threshMax ){
+                if(RPM_VALUES[i] > threshMax){
+                if(RPM_VALUES[i] > maxVal){
+                    maxVal = RPM_VALUES[i];
+                    WHEEL = i;
+                }
+                // printf("Slippage Detected: Wheel %d\r\n", i);
+                push(WHEELS_WITHOUT_TRACTION,i);
+            }
+            else{ push(WHEELS_WITH_TRACTION,i); }
+        }
+    linearTraction(THROTTLE_INPUT);
+        }
+    }
```

Comment:
`RPM_VALUES[]` is no longer saved in an array and is instead held in an object as per https://github.com/Gcerpa01/R.E.S.S.T/blob/a780daf932267df8bfbca05a96f845e34d7b234a/STM32Project/Core/Inc/basic_helpers.h#L30

What you can do is make the tuple an external tuple in 
https://github.com/Gcerpa01/R.E.S.S.T/blob/a780daf932267df8bfbca05a96f845e34d7b234a/STM32Project/Core/Src/controller_conversion.c#L61-L62

and access the RPM of those structs directly

Reply:
Actually they are already global accessible tuples so you can just access them instead, no need to declare again